File: 409-longest-palindrome/409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char, int> m;
        for(auto i : s)
        {
            m[i]++;
        }
    int sum = 0;
    int odd = 0;
    for(auto i : m)
    {
        if(i.second % 2 == 1)
        {
            odd++;
            sum += i.second - 1;
        }
        else
        {
            sum += i.second;
        }
    }
    if(odd != 0)
    {
        return sum + 1;
    }
    return sum;
    }
};

File: 409-longest-palindrome/409-longest-palindrome.java
class Solution {
   public int longestPalindrome(String s) {
        Map<Character,Integer> hm= new HashMap<>();
        char[] c= s.toCharArray();
        int freq=0,odd=0;
        int flag=0;
        for(char i: c)
        {
         hm.put(i,hm.getOrDefault(i,0)+1);
            freq++;
        }
       for (Map.Entry<Character,Integer> i : hm.entrySet()) {
           if(i.getValue() % 2 == 1)
           {
               flag = 1;
               odd++;
           }
       }
        if(flag==1)
        {
            return freq-(odd-1);
        }
        return freq;
        
    }
}